### PR TITLE
Fix: Neurons Route Test

### PR DESCRIPTION
**Pull request by lmuntanerDfinity**
# Motivation

Neurons Route test was passing for the wrong reasons.

# Changes

* Add `data-test` to Card, and NeuronCard to identity cards in tests.
* Changes in Neurons route test.
* Reuse the neurons subscribe mock in multiple tests.
* Fixed some style issues.

# Tests

All changes are in the tests.

### CODE REVIEW

**Comment by lmuntanerDfinity** [comment]
@peterpeterparker @mstrasinskis please review

**Review by peterpeterparker — CHANGES_REQUESTED** [review]

File: frontend/svelte/src/lib/components/neurons/NeuronCard.svelte [hunk 1]
```diff
@@ -15,14 +15,28 @@
...
+    neuron.state === NeuronState.DISSOLVING &&
+    neuron.fullNeuron !== undefined &&
+    neuron.fullNeuron.dissolveState !== undefined &&
+    "WhenDissolvedTimestampSeconds" in neuron.fullNeuron.dissolveState
+      ? neuron.fullNeuron.dissolveState.WhenDissolvedTimestampSeconds
+      : undefined;
 </script>
 
-<Card {role} on:click {ariaLabel}>
+<Card {role} on:click {ariaLabel} dataTest="neuron-card">
```
[the remark this answers is not in the document]
Comment (lmuntanerDfinity) [reply frontend/svelte/src/lib/components/neurons/NeuronCard.svelte:39]: The problem was that before I was relying on the `a`, and it was failing. Using a more specific but still generic selector can have the same problem.

Shall I create an issue to look into this?

File: frontend/svelte/src/lib/components/ui/Card.svelte [hunk 2]
```diff
@@ -1,13 +1,20 @@
 <script lang="ts">
   export let role: "link" | "button" | undefined = undefined;
   export let ariaLabel: string | undefined = undefined;
+  export let dataTest: string | undefined = undefined;
 
   let clickable: boolean = false;
 
-  $: clickable = ["button", "link"].includes(role);
+  $: clickable = role !== undefined ? ["button", "link"].includes(role) : false;
```
[the remark this answers is not in the document]
Comment (lmuntanerDfinity) [reply frontend/svelte/src/lib/components/ui/Card.svelte:8]: Yes

File: frontend/svelte/src/tests/routes/Neurons.spec.ts [hunk 3]
```diff
@@ -41,10 +59,14 @@ describe("Neurons", () => {
...
-  it("should render a NeuronCard", () => {
+  it("should render a NeuronCard", async () => {
     const { container } = render(Neurons);
 
-    const anchor = container.querySelector("a");
+    // Wait until `isLoading = false`
+    await tick();
+    await tick();
+    await tick();
+    const anchor = container.querySelector('[data-test="neuron-card"]');
```
[the remark this answers is not in the document]
Comment (lmuntanerDfinity) [reply frontend/svelte/src/tests/routes/Neurons.spec.ts:69]: good idea!

File: frontend/svelte/src/lib/components/neurons/NeuronCard.svelte [hunk 4]
```diff
@@ -15,14 +15,28 @@
...
+      ? true
+      : !neuron.fullNeuron?.isCurrentUserController;
+  let neuronICP: ICP | undefined;
+  $: neuronICP =
+    neuron.fullNeuron !== undefined
+      ? ICP.fromE8s(neuron.fullNeuron.cachedNeuronStake)
+      : undefined;
+  let dissolvingTime: bigint | undefined;
+  $: dissolvingTime =
+    neuron.state === NeuronState.DISSOLVING &&
```
[the remark this answers is not in the document]
Comment (lmuntanerDfinity) [reply frontend/svelte/src/lib/components/neurons/NeuronCard.svelte:28]: I create an issue and added a comment above these lines.

**Comment by lmuntanerDfinity** [comment]
@peterpeterparker @mstrasinskis ready for another review

File: frontend/svelte/src/lib/components/neurons/NeuronCard.svelte (hunk 1 above)
[the remark this answers is not in the document]
Comment (lmuntanerDfinity) [reply frontend/svelte/src/lib/components/neurons/NeuronCard.svelte]: Removed data-test dependence

**Review by peterpeterparker — APPROVED** [review]
LGTM

**Review by mstrasinskis — APPROVED** [review]
Thanks!

**Pull request merged**